### PR TITLE
Add ability to uninstall plugins

### PR DESCRIPTION
**Pull request by fatkodima**
Reopened #3438 

@deivid-rodriguez You are right (https://github.com/rubygems/rubygems/pull/3438#discussion_r403256897). Updated with your suggestion.

Closes #3438

### CODE REVIEW

File: bundler/lib/bundler/plugin.rb [hunk 1]
```diff
@@ -47,6 +47,32 @@ def install(names, options)
...
+            Bundler.rm_rf(index.plugin_path(name))
+            index.unregister_plugin(name)
+            Bundler.ui.info "Uninstalled plugin #{name}"
+          else
+            Bundler.ui.error "Plugin #{name} is not installed \n"
+          end
+        end
+      else
+        Bundler.ui.info "No plugins installed"
+      end
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/plugin.rb:73]: I wonder if we should simplify this method to do something symmetric to what the `install` method:

* Extract an `Plugin:Uninstaller` class which takes care of the actual file removal and returns the list of uninstalled plugins. Maybe this class could be reused for the `rescue` block logic of the `install` method?

* Move the unregistration of the uninstalled plugins returned by the `Uninstaller` to an `unsave_plugins` helper.

Do you think this refactoring is worth doing?

File: bundler/lib/bundler/plugin.rb [hunk 2]
```diff
@@ -47,6 +47,32 @@ def install(names, options)
       Bundler.ui.error "Failed to install plugin #{name}: #{e.message}\n  #{e.backtrace.join("\n ")}"
```
Comment (deivid-rodriguez) [inline bundler/lib/bundler/plugin.rb:47]: I was reading this block to see if some of the "uninstallation logic" could/should be reused here, and I'm wondering... where is the `name` variable being defined? Is this actually working/neeed? :thinking:

I know this is not your code, but if you're willing to double check this, I appreciate it.

Comment (fatkodima) [reply bundler/lib/bundler/plugin.rb:47]: This is needed, but not working. In eb5873fc0e method signature was changed, but this line stayed the same.
I think it is better to fix this in a separate pr.

File: bundler/lib/bundler/plugin.rb (hunk 1 above)
Comment (fatkodima) [reply bundler/lib/bundler/plugin.rb:73]: Imo, this method is pretty simple right now and and we have better control with logging messages. I'd left it as it is for now.

Comment (deivid-rodriguez) [reply bundler/lib/bundler/plugin.rb:73]: Ok, let's leave it for now.

File: bundler/lib/bundler/plugin.rb (hunk 2 above)
Comment (deivid-rodriguez) [reply bundler/lib/bundler/plugin.rb:47]: I opened #3484 to track it.

**Comment by deivid-rodriguez** [comment]
Let's do it!

Thanks @fatkodima!

**Pull request merged**